cnxdb/archive-sql/schema/triggers.sql

```diff
@@ -248,54 +248,67 @@ CREATE TRIGGER ruleset_trigger
 
 -- recipe maintenace triggers
 
-CREATE OR REPLACE FUNCTION update_default_recipes() RETURNS trigger AS '
+CREATE OR REPLACE FUNCTION update_default_recipes() RETURNS trigger AS $$
 BEGIN
-  IF (TG_OP = ''INSERT'' OR TG_OP = ''UPDATE'') AND
+  IF (TG_OP = 'INSERT' OR TG_OP = 'UPDATE') AND
           NEW.revised >= ((SELECT revised FROM print_style_recipes
               WHERE print_style = NEW.print_style ORDER BY revised DESC LIMIT 1)
-              UNION ALL VALUES (NEW.revised) LIMIT 1)
+              UNION ALL VALUES (NEW.revised) LIMIT 1) -- avoid compare to NULL
   THEN
       DELETE FROM default_print_style_recipes WHERE print_style = NEW.print_style;
       INSERT into default_print_style_recipes (
-        print_style, tag, fileid, recipe_type, revised)
+        print_style, title, tag, commit_id, fileid, recipe_type, revised)
   	VALUES (
-        NEW.print_style, NEW.tag, NEW.fileid, NEW.recipe_type, NEW.revised);
+        NEW.print_style, NEW.title, NEW.tag, NEW.commit_id, NEW.fileid, NEW.recipe_type, NEW.revised);
   END IF;
 
-  IF TG_OP = ''UPDATE'' THEN
-      UPDATE default_print_style_recipes SET
-        fileid=NEW.fileid,
-        recipe_type=NEW.recipe_type,
-        NEW.revised)
-        WHERE print_style=NEW.print_style AND tag=NEW.tag;
+  IF TG_OP = 'UPDATE' THEN
+    PERFORM 1 from modules where recipe = OLD.fileid;
+    IF FOUND and NEW.fileid != OLD.fileid THEN
+        RAISE WARNING 'Cannot change recipe file: recipe in use';
+        RETURN NULL;
+    END IF;
```

Comment:
This block is preventing a recipe from being overwritten if it is in use somewhere, correct?

Reply:
Right - similar code in the BEFORE DELETE trigger.